README.md
# bga-forbiddenisland


misc/RELEASE_NOTES.txt
Oct 06
* bug with flood deck cardcount
* update sounds

Oct 04
* stock for rotated display
* ELO adjustments
* steamline give card for 2-players
* sound effects
* more island maps

Sep 28
* flood card rotated display animation
* player color adjustments
* bug in rescue pawn fail game logic
* score based on difficulty (power 2)

Sep 24
* #25264: "DIVER CAN MOVE TO THE TILE WHERE HE IS STANDING"
* minor usability tweaks
* #25240: "WATER LEVEL MARKER IS WRONG ON TABLET"

Sep 23
* final usability feedback for beta release
* remove non-actionable buttons from status bar

Sep 20
* water background w/ changing color

Sep 13
* tile tooltips

Aug 30
* #23190: "TREASURE DID NOT DISPLAY AS BEEN CAPTURED"
* add water level to final table

Aug 27
* #23085: "CAN'T END TURN"; #23122: "DISCARDING VISUAL BUG"
* support for large screens

Aug 22
* #21997: "DISCARD PILE DIDN'T REFRESH AFTER SHUFFLE"
* changed to use stock in place of zone for card displays
* #20258: "HAND CARDS DISPLAY BUGS"
* #20300: "F5 TO REFRESH CARDS IN HAND"
* #22436: "CAN'T GIVE A CARD TO A PLAYER THAT HAS ZERO CARDS"

Aug 01
* add player symbols to boards
* completed logic for Zombie turns
* #20295: "NO HOVERING TEXT FOR SECOND DISCARD PILE"
* tooltip for possible moves vs shore_up
* logic of undo or special after 3rd action
* added continue state before drawing cards to allow for special actions

Jul 26
* mobility scaling
* larger 147px width tiles and cards
* #20705: "SUGGESTION TO ALLOW PLAYERS TO KNOW HOW MANY CARDS ARE LEFT IN THE DECK"
* #21674: "UNABLE TO FLY TO SAFETY WHEN TILE SUNK"
* #20516: "THE GAME MAP DOESN'T SHOW UP ENTIRELY ON MOBILE"
* #20509: "BOARD IS BEHIND PLAYER PANEL (MOBILE/TABLET SCREENS)"
* #21354: "CAPITALISED OR NOT?"

Jul 17
* incorrect name posted in log for special action
* #20952: "GAME IS STUCK ON RESCUEING PAWN"
* #20922: "GAME STUCKED WHEN TILE WHERE I WAS ON SUNCK"
* #20218: "PLAY OUT OF TURN & GAMELOG - INCORRECT PLAYER NAME"
* #21036: "IF 4 PLAYERS ARE ON AN ISLAND SPACE, IT SHOWS INCORRECTLY ON MOBILE/TABLET"

Jul 11
* #20591: "COULD NOT SELECT TILE FOR SANDBAGS AS PART OF A DISCARD"

Jul 08
* #20474: "GAME STUCKED"
* #20345: "GAME DOSN'T LOAD"
* #20411: "GAME LOCKED AND WE HAVE HAD TO ABBANDON"

Jul 07
* fix pilot start tile
* #20232: "GAME HUNG" - issue with possibleAction search
* show remainging actions for other players 


Jul 06
* #20221: "GAME ENDED WHEN A TREASURE TILE JUST FLOODED"
* thething - display scaling
* #20217: "GAME IS OVERLAPPING WITH GAME LOG"
* #20384: "DIVER RULES STILL NOT RESPECTED"

Jul 05
* #20301: "NAVIGATOR POWER DOESN'T WORK"
* #20310: "DIVER RULES NOT RESPECTED"
* End game result table
* #20283: "COULD NOT SELECT MANY TILES FOR HELICOPTER LIFT DESTINATION"
* added new island maps

Jul 04
*  #20230 - "Could not claim treasure from flooded tile"
* added debugLoadReport
* multiple cards showing selected
* #20220 - "Need to press F5 to shore up near-by tile"
* fixed Navigator issue
* implemented End Turn instead of skip

Jul 03
* player feedback on capture treasure failure
* player board order
* confirm on discard
* flip player name on player board, add player color
* fix issue with spectator

stats.inc.php
<?php

/**
 *------
 * BGA framework: © Gregory Isabelli <gisabelli@boardgamearena.com> & Emmanuel Colin <ecolin@boardgamearena.com>
 * forbiddenisland implementation : © Jeff DiCorpo <jdicorpo@gmail.com>
 *
 * This code has been produced on the BGA studio platform for use on http://boardgamearena.com.
 * See http://en.boardgamearena.com/#!doc/Studio for more information.
 * -----
 *
 * stats.inc.php
 *
 * forbiddenisland game statistics description
 *
 */

/*
    In this file, you are describing game statistics, that will be displayed at the end of the
    game.
    
    There are 2 types of statistics:
    _ table statistics, that are not associated to a specific player (ie: 1 value for each game).
    _ player statistics, that are associated to each players (ie: 1 value for each player in the game).

    Statistics types can be "int" for integer, and "float" for floating point values.
    
    Once you defined your statistics there, you can start using "initStat", "setStat" and "incStat" method
    in your game logic, using statistics names defined below.
*/

//   !! It is not a good idea to modify this file when a game is running !!


$stats_type = array(

    // Statistics global to table
    "table" => array(

        "turns_number" => array("id"=> 10,
                    "name" => totranslate("Number of turns"),
                    "type" => "int" ),
        "players_number" => array("id"=> 11,
                    "name" => totranslate("Number of players"),
                    "type" => "int" ),
        "beginners_number" => array("id"=> 12,
                    "name" => totranslate("Number of beginner players"),
                    "type" => "int" ),
        "difficulty" => array("id"=> 13,
                    "name" => totranslate("Game difficulty"),
                    "type" => "int" ),
        "island_map" => array("id"=> 14,
                    "name" => totranslate("Island map played"),
                    "type" => "int" ),
        "tiles_flooded" => array("id"=> 15,
                    "name" => totranslate("Number of tiles flooded"),
                    "type" => "int" ),
        "tiles_sunk" => array("id"=> 16,
                    "name" => totranslate("Number of tiles sunk"),
                    "type" => "int" ),
        "tiles_shored_up" => array("id"=> 17,
                    "name" => totranslate("Number of tiles shored up"),
                    "type" => "int" ),
        "treasures_captured" => array("id"=> 18,
                    "name" => totranslate("Treasures captured"),
                    "type" => "int" ),
        "water_level" => array("id"=> 19,
                    "name" => totranslate("Water level"),
                    "type" => "int" ),
        "players_won" => array("id"=> 20,
                    "name" => totranslate("Players won"),
                    "type" => "bool" ),
    ),
    
    // Statistics existing for each player
    "player" => array(

        "turns_number" => array("id"=> 100,
                    "name" => totranslate("Number of turns"),
                    "type" => "int" ),
        "adventurer" => array("id"=> 101,
                    "name" => totranslate("Adventurer played"),
                    "type" => "int" ),
        "move" => array("id"=> 102,
                    "name" => totranslate("Move actions"),
                    "type" => "int" ),
        "shore_up" => array("id"=> 103,
                    "name" => totranslate("Shore up actions"),
                    "type" => "int" ),
        "give_card" => array("id"=> 104,
                    "name" => totranslate("Give treasure card actions"),
                    "type" => "int" ),
        "capture" => array("id"=> 105,
                    "name" => totranslate("Treasures captured"),
                    "type" => "int" ),
        "skip" => array("id"=> 106,
                    "name" => totranslate("Actions skipped"),
                    "type" => "int" ),
        "discard" => array("id"=> 107,
                    "name" => totranslate("Treasure cards discarded"),
                    "type" => "int" ),
        "sandbags" => array("id"=> 108,
                    "name" => totranslate("Sandbag cards played"),
                    "type" => "int" ),
        "heli_lift" => array("id"=> 109,
                    "name" => totranslate("Helicopter Lift cards played"),
                    "type" => "int" ),
        "pilot" => array("id"=> 110,
                    "name" => totranslate("Pilot actions"),
                    "type" => "int" ),
        "navigator" => array("id"=> 111,
                    "name" => totranslate("Navigate actions"),
                    "type" => "int" ),
        "bonus_shoreup" => array("id"=> 112,
                    "name" => totranslate("Engineer bonus shore ups"),
                    "type" => "int" ),
    
    ),

    "value_labels" => array(
        13 => array(
            1 => totranslate("Novice"),
            2 => totranslate("Normal"),
            3 => totranslate("Elite"),
            4 => totranslate("Legendary"),
        ),
        14 => array (
            1 => totranslate('Forbidden Island'),
            2 => totranslate('Treasure Island'),
            3 => totranslate('Island of Shadows'),
            4 => totranslate('Volcano Island'),
            5 => totranslate('Island of Death'),
            6 => totranslate('Bay of Gulls'),
            7 => totranslate('Bone Island'),
            8 => totranslate('Coral Reef'),
            9 => totranslate('Skull Island'),
            10 => totranslate('Bridge of Horrors'),
            11 => totranslate('Atoll of Decisions'),
            12 => totranslate('Arch of Fate'),
        ),
        101 => array (
            1 => totranslate('Engineer'),
            2 => totranslate('Pilot'),
            3 => totranslate('Navigator'),
            4 => totranslate('Explorer'),
            5 => totranslate('Diver'),
            6 => totranslate('Messenger'),
        ),
    )

);
